pymeasure/instruments/keysight/keysightE3631A.py

```diff
@@ -0,0 +1,133 @@
+
+
+class VoltageChannel(Channel):
+    """Implementation of a power supply base class channel"""
+
+    voltage_setpoint = Channel.control(
+        "INST:NSEL {ch};:VOLT?",
+        "INST:NSEL {ch};:VOLT %g",
+        """Control the output voltage of this channel, range depends on channel.""",
+        validator=strict_range,
+        values=[0, 25],
+        dynamic=True,
+    )
+
+    current_limit = Channel.control(
+        "INST:NSEL {ch};:CURR?",
+        "INST:NSEL {ch};:CURR %g",
+        """Control the current limit of this channel, range depends on channel.""",
+        validator=strict_range,
+        values=[0, 1],
+        dynamic=True,
+    )
+
+    voltage = Channel.measurement(
+        "INST:NSEL {ch};:MEAS:VOLT?",
+        """Measure actual voltage of this channel.""",
+        dynamic=True
+    )
+
+    current = Channel.measurement(
+        "INST:NSEL {ch};:MEAS:CURR?",
+        """Measure the actual current of this channel."""
+    )
+
+
+class KeysightE3631A(Instrument):
+    """ Represents the Keysight E3631A Triple Output DC Power Supply
+    interface for interacting with the instrument.
+
+    .. code-block:: python
+
+        supply = KeysightE3631A(resource)
+        supply.ch_1.voltage_setpoint=10
+        supply.ch_1.current_setpoint=0.1
+        supply.ch_1.output_enabled=True
+        print(supply.ch_1.voltage)
+    """
+
+    ch_1 = Instrument.ChannelCreator(VoltageChannel, 1)
+
+    ch_2 = Instrument.ChannelCreator(VoltageChannel, 2)
+
+    ch_3 = Instrument.ChannelCreator(VoltageChannel, 3)
+
+    def __init__(self, adapter, name="Keysight E3631A", **kwargs):
+        super().__init__(
+            adapter, name, **kwargs
+        )
+        self.channels[1].voltage_setpoint_values = [0, 6]
+        self.channels[1].current_limit_values = [0, 5]
+        self.channels[3].voltage_setpoint_values = [0, -25]
+
+    tracking_enabled = Instrument.control(
+        ":OUTP:TRAC?",
+        ":OUTP:TRAC %s",
+        """Control whether the power supply operate in the track mode (boolean)""",
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    output_enabled = Instrument.control(
+        "OUTPut?",
+        "OUTPut %d",
+        """Control whether the channel output is enabled (boolean).""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={True: 1, False: 0},
+        dynamic=True,
+    )
+
+    def ch(self, channel_number):
+        """Get a channel from this instrument.
+
+        :param: channel_number:
+            int: the number of the channel to be selected
+        :type: :class:`.VoltageChannel`
+
+        """
+        if channel_number == 1:
+            return self.ch_1
+        elif channel_number == 2:
+            return self.ch_2
+        elif channel_number == 3:
+            return self.ch_3
+        else:
+            raise ValueError("Invalid channel number. Must be 1, 2, 3.")
+
```

Comment:
That method is superfluous as you can access the channels easily:
```python
inst.channels[1] == inst.ch_1
inst.channels[2] == inst.ch_2
inst.channels[3] == inst.ch_3
```

All three equalities are true as all Channels are added to the `channels` dictionary, where their `id` is the corresponding key.

Reply:
Yes I know. This method is used in the test_keysightE3631A_with_device.py to dynamically pass the good channel in the test methods i.e `power_supply.ch(chn).voltage_setpoint == voltage`. I don't find another way to do that

Reply:
what about `power_supply.channels[chn].voltage_setpoint == voltage`?

Reply:
obviously...change has been made. Thank you.